src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -27,9 +30,9 @@
              comidi/routes->handler)
          settings
          path
-         wrap-with-authorization-check
-         puppet-version)))
-   context)
+         (fn [request] (wrap-with-authorization-check request {:oid-map oid-mappings}))
```

Comment:
Hmm, that's also a valid option. I think I'd probably go with the first option though. It's a bit weirder, but adding in another service adds in another layer of complexity, and it will make it a bit harder to upgrade as existing users will need to add the new service to their bootstrap configuration files if they've made any changes.

I guess that's probably not a huge deal though.